### PR TITLE
Reworked the unix launcher

### PR DESCRIPTION
**Pull request by kurumushi**
I fixed up the shell script a bit (ran it through shellcheck) for gotchas and posix compliance.

I also changed the terminal detection to make it a bit more intuitive and easier to debug.

This may solve #639 as well.

### CODE REVIEW

**Comment by kurumushi** [comment]
Forgot to add that this is HIGHLY untested, it works for me, but should be tested thoroughly before being put into stable.

**Comment by toastal** [comment]
You may want to attribute yourself in the top comment...

**Comment by toastal** [comment]
The script seems to work for me. There's a `--no-terminal` option. Can this be exposed to this script as well?

**Comment by kurumushi** [comment]
>LAUNCHER="./StardewModdingAPI.bin.x86 $*"

Any extra arguments should already be passed on to the launcher.

**Comment by kurumushi** [comment]
If we could assume that every terminal would work with TERM set to xterm, it might be possible to change the command to 'env TERM=xterm ./StardewModdingAPI.bin $*' and avoid all this mess.

Also just for reference, this is all related to mono/mono#6752

File: src/SMAPI.Installer/unix-launcher.sh [hunk 1]
```diff
@@ -62,36 +62,58 @@ else
...
-        # Terminator converts -e to -x when used through x-terminal-emulator for some reason (per
-        # `man terminator`), which causes an "unable to find shell" error. If x-terminal-emulator
-        # is mapped to Terminator, invoke it directly instead.
-        if [[ "$(readlink -e $(which x-terminal-emulator))" == *"/terminator" ]]; then
-            terminator -e "sh -c 'TERM=xterm $LAUNCHER'"
-        else
-            x-terminal-emulator -e "sh -c 'TERM=xterm $LAUNCHER'"
+    # First let's try xterm (best compatiblity) or find a sensible default
+    # Setting TERMINAL should let you override this at the commandline for easier testing of other terminals.
+    for terminal in  "$TERMINAL" xterm x-terminal-emulator kitty terminator xfce4-terminal gnome-terminal konsole terminal termite; do
```
Comment (toastal) [inline src/SMAPI.Installer/unix-launcher.sh]: extra space after `in`

File: src/SMAPI.Installer/unix-launcher.sh [hunk 2]
```diff
@@ -62,36 +62,58 @@ else
...
+        if $COMMAND "$terminal" 2>/dev/null; then
+            # Find the true shell behind x-terminal-emulator
+            if [ "$(basename "$(readlink -ef which "$terminal")")" != "x-terminal-emulator" ]; then
+                    export LAUNCHTERM=$terminal
```
Comment (toastal) [inline src/SMAPI.Installer/unix-launcher.sh]: everything else in this file is 4-spaced, but this is 8

**Comment by Pathoschild** [comment]
Thanks! (I'm backporting some changes to a 2.11.2 release, so you can ignore the conflict being shown. I'll take a look for the 3.0 release after that.)

**Comment by toastal** [comment]
Also of note, I tested on my other machine with Openbox and this fixed my issue there as well

**Comment by kurumushi** [comment]
Seems like linux users are ok, but definitely need to have someone with OSX check to make sure it works. I didn't change a lot in the OSX portion of the script, but I'm not sure how the shell/terminal will like the changes I did make.

**Comment by Pathoschild** [comment]
Thanks! GitHub didn't like the force-push to fix the merge conflict, but I manually merged it into `develop`. I'll do some testing with MacOS users to make sure too.

**Pull request closed without merging**